src/algorithms/local_search/local_search.cpp

```diff
@@ -147,20 +314,33 @@ void LocalSearch<Route,
                  RouteExchange>::try_job_additions(const std::vector<Index>&
                                                      routes,
                                                    double regret_coeff) {
+
   bool job_added;
-  std::vector<Gain> best_costs;
-  std::vector<Index> best_ranks;
-  std::vector<Index> best_pickup_ranks;
-  std::vector<Index> best_delivery_ranks;
+    
+  std::vector<std::vector<RouteInsertion>> route_job_insertions;
+
+  for (std::size_t i = 0; i < routes.size(); ++i) {
+      //TODO: tweak such that the vectors don't include all jobs
```

Comment:
Have you considered mapping `RouteInsertion` objects with unassigned ranks as key (`j` value in the next loop)? That would make `route_job_insertions` a `std::vector<std::unordered_map<Index, RouteInsertion>>`.

If this is less efficient than a plain `std::vector`, I think we can leave with the extra non-used ranks for assigned jobs.

Reply:
I tried dropping in a `std::unorderd_map` as you suggested but it increased the running time of the Li&lim instances I tried by 20-30%, that's why I removed the comment and left the vector as is.